src/include/splash/basetypes/ColTypeString.hpp

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright 2013-2015 Felix Schmitt, Axel Huebl
+ *
+ * This file is part of libSplash.
+ *
+ * libSplash is free software: you can redistribute it and/or modify
+ * it under the terms of of either the GNU General Public License or
+ * the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * libSplash is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License and the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * and the GNU Lesser General Public License along with libSplash.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef COLTYPESTRING_H
+#define	COLTYPESTRING_H
+
+#include "splash/CollectionType.hpp"
+
+namespace splash
+{
+    /** Variable and fixed length strings
+     *
+     * Since C-strings are NULL-terminated, len must be +1
```

Comment:
@f-schmitt-zih I tried to implement the change but it looks extremely odd in places [next to the fixed-length c-strings](https://github.com/ax3l/libSplash/blob/7752518ea375493160444f0f1ac7880891ddb670/tests/Parallel_AttributesTest.cpp#L104-L105).

actually, we should not compare it to `strlen()` but to the c-"constructor" of arrays such as the mentioned `c[5]` above. are you sure we want to prefer an implicit "+1"?

examples:

```c++
const char c[4] = {"ABCD"}; // valid C and invalid C++
const char d[5] = {"ABCD"}; // valid C and valid C++
```